perf/src/deduper.rs

```diff
@@ -48,16 +51,14 @@ impl<const K: usize> Deduper<K> {
         }
     }
 
-    // Returns true if the packet is duplicate.
+    // Returns true if the data is duplicate.
     #[must_use]
     #[allow(clippy::integer_arithmetic)]
-    fn dedup_packet(&self, packet: &Packet) -> bool {
-        // Should not dedup packet if already discarded.
-        debug_assert!(!packet.meta().discard());
+    pub fn check_duplicate(&self, data: &T) -> bool {
```

Comment:
```suggestion
    pub fn is_duplicate(&self, data: &T) -> bool {
```

"check" doesn't particularly say anything about what this method does

Reply:
The problem with `is_duplicate` is that it implies the function is "pure".
Whereas this function has side-effects, and calling it on a value twice returns different results.

Reply:
renamed to `is_duplicate`.

Reply:
reverted to what it was before: `dedup`.